// @tesla/validation/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
exports.vatNumber = exports.password = exports.CreditCard = exports.validatePhone = exports.validateEmail = exports.validatePostalCode = void 0;
var tslib_1 = require("tslib");
var postalCode_1 = require("./postalCode");
Object.defineProperty(exports, "validatePostalCode", {
    enumerable: true,
    get: function() {
        return tslib_1.__importDefault(postalCode_1).default;
    }
});
var email_1 = require("./email");
Object.defineProperty(exports, "validateEmail", {
    enumerable: true,
    get: function() {
        return tslib_1.__importDefault(email_1).default;
    }
});
var phone_1 = require("./phone");
Object.defineProperty(exports, "validatePhone", {
    enumerable: true,
    get: function() {
        return tslib_1.__importDefault(phone_1).default;
    }
});
var creditCard_1 = require("./creditCard");
Object.defineProperty(exports, "CreditCard", {
    enumerable: true,
    get: function() {
        return tslib_1.__importDefault(creditCard_1).default;
    }
});
var password_1 = require("./password");
Object.defineProperty(exports, "password", {
    enumerable: true,
    get: function() {
        return tslib_1.__importDefault(password_1).default;
    }
});
var internalVatNumber = tslib_1.__importStar(require("./vatNumber"));
exports.vatNumber = internalVatNumber;
